doc/Language/exceptions.pod6

```diff
@@ -84,9 +93,15 @@ methods apply to C<$_>, which holds the C<Exception> within the C<CATCH> block.
          }
     }
 
-Note that the match target is a role. To allow user defined exceptions to match
-in the same manner, they must implement the given role. Just existing in the
-same namespace will look alike but won't match in a C<CATCH> block.
+Note that the match target is a role. To allow user defined exceptions to
+match in the same manner, they must implement the given role. Just existing
+in the same namespace will look alike but won't match in a C<CATCH> block.
+
+Note that the C<CATCH> block semantics apply to the B<entire> lexical scope
+in which it is defined, B<regardless> of where it is defined inside that
+lexical scope.  It is therefore advised to put any C<CATCH> block at the
+starte of the lexical scope to which they apply so that the casual reader
+of the code can immediately see that there is something special going on.
```

Comment:
This clarification is on point, but it might be better right after the example, or alternatively providing another example.